### ISSUE TITLE
Fit and finish for Insert Item...

### ISSUE THREAD

**Issue by StephenWeatherford** [microsoft/vscode-azurearmtools#670]
- [x] P2: Doubled "the the" in the messages "Please type the the value of ..."
- [x] P2: We should add "Do not show again" functionality to these messages -> I'll do this as part of another PR
- [x] P1: After inserting, the location of the cursor should get revealed (currently the entire span gets revealed?)
- [x] ~~P3: Should we add validation (e.g. function name already exists, invalid names)?~~
- [x] P2: Now that we have a simple icon button to sort individual sections, it's a little disconcerting if I click, say, on the parameters' sort button and it says "Done sorting template!".  It would be better to be more specific "Parameters were sorted".  Ideally it would say "No parameters needed sorting" if appropriate.

**Comment by nilshedstrom** [comment]
###  After inserting, the location of the cursor should get revealed 
I try to reveal the position where the text was inserted 
```
        let pos = textEditor.document.positionAt(index);
        await textEditor.edit(builder => builder.insert(pos, text));
        textEditor.revealRange(new vscode.Range(pos, pos), vscode.TextEditorRevealType.Default);
```
For parameters I do not change the cursor. What should I reveal then?

What are the name validation rules for parameters, variables, functions, namespaces and outputs?

**Comment by StephenWeatherford** [comment]
@neilpeterson What do you think about validation?  Is it worth the effort?

**Comment by neilpeterson** [comment]
I am not aware of validation rules for parameter, variable, function, or output names. Checking both [this doc](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-syntax), and [this doc](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md), the only requirements I've found is "_Must be a valid JavaScript identifier_" for both parameters and outputs.

```
Must be a valid JavaScript identifier
```

As for validation of duplicate names. We do throw an error on duplicate names for parameters, variabels, and outputs. So catching this at insert time would add some value.

![image](https://user-images.githubusercontent.com/7844635/81141523-c3ce6e80-8f21-11ea-97ac-69dc1917afb0.png)

My opinion is low priority assuming it's a simple addition. We can also let it sail without and adjust if needed / desired at a later time.

**Comment by StephenWeatherford** [comment]
@nilshedstrom Sorry for the delay.  Re "After inserting, the location of the cursor should get revealed (currently the entire span gets revealed?)", here's an example:
![reveal](https://user-images.githubusercontent.com/6913354/81854335-a77a9500-9512-11ea-9365-b6e1a2b7fbf8.gif)
Without looking at the code, it looks like you're revealing the entire inserted portion?  Which is fine unless you're also setting the cursor, in which case you should reveal the cursor location.
Also, I'd suggest trying TextEditorRevealType.InCenterIfOutsideViewport.

**Comment by neilpeterson** [comment]
@nilshedstrom can we consider removing the sort icon from individual elements and only present it on the root level sections?

![image](https://user-images.githubusercontent.com/7844635/82759879-da3f4b80-9da4-11ea-8a56-7a5b356c4119.png)

**Comment by nilshedstrom** [comment]
@neilpeterson 
![image](https://user-images.githubusercontent.com/785949/82766170-17dfbb00-9e1d-11ea-9f51-d9b306859cbd.png)
Fixed that. Did you want to remove the item from the context menu too?

**Comment by StephenWeatherford** [comment]
> @nilshedstrom can we consider removing the sort icon from individual elements and only present it on the root level sections?

Yes, please.

> Did you want to remove the item from the context menu too?

No, the context menu items are more accessible/discoverable for someone with vision issues.

**Comment by StephenWeatherford** [comment]
Last two items fixed by #727
